spec/index.bs

```diff
@@ -1100,7 +1101,28 @@ To <dfn abstract-op export>get Trusted Types-compliant attribute value</dfn> on
 
    If the algorithm threw an error, rethrow the error.
 
-Issue: This algorithm should account for event handler attributes. See https://github.com/w3c/trusted-types/issues/474
+## Get Trusted Type data for attribute ## {#get-trusted-type-data-for-attribute}
+To <dfn abstract-op>Get Trusted Type data for attribute</dfn> given |element|, |attribute|, |attributeNs|, perform the following steps:
+
+1. Let |data| be null.
+1. If |attributeNs| is null, and |attribute| is the name of an [=event handler content attribute=], then:
```

Comment:
I'm not really sure how you envision this lookup to work. There are many event handler content attributes that are not applicable to elements. And some only apply to a specific element.

Reply:
> There are many event handler content attributes

I don't fully understand what you mean by this? What event handler content attribute doesn't apply to an element?
Do you mean legacy ones no longer applicable or something like that?

> And some only apply to a specific element.

So for the purposes of trusted types I don't think we mind being overly restrictive here, it seems fine to consider them independent of the element (this is the case in Chromium's implementation)?

For context the plan in WebKit is to use `HTMLElement::eventNameForEventHandlerAttribute` with a null check.

Reply:
`onreadystatechange`?

I'm not sure we have a specification concept for `eventNameForEventHandlerAttribute`.

Reply:
Chrome enforces the TT on that attribute for a div element for example, so that aspect is fine.

Reply:
If I understand correctly the issue would be that you require a trusted object to set an inline event listener that is "safe" for a given element. Given inline event listeners are bad practice and a strict csp would already block them from actually running I don't think blocking "harmless" but otherwise useless (like are people actually going to be trying to set onreadystatechange on an arbitrary element?) attributes from being set from a string is fine?

Reply:
onloadend also requires a trusted script to be set on div

Reply:
What about `onblah`?

My worry here is about the hook not being defined in sufficient detail for an interoperable implementation. Implementing "is an event handler content attribute" requires knowing about all event handler attributes in existence, not just those defined in HTML.

Reply:
`onblah` is allowed. I *think* it's using IDL to generate the list from `attribute EventHandler`s `onrepeat` for example which as far as I can tell is only SVG related is being blocked.

Reply:
Given we can't use the on* pattern what is the next best steps here?

Reply:
Both Chrome and WebKit have a way to look up EventHandlers from IDL which gives us the coverage we need to ensure stuff is protected. Idk if that corresponds to the current spec text, or how we would spec that?

The key bit is that all sinks for a given browser are included in that browser, which the spec as written I think covers?

Reply:
I think we should have an actual list of event handler attributes we care about (those on elements), which would be a subset of all event handler attributes. That's also what the Sanitizer API would need.

I don't think we want to do something with event handler attributes that only have meaning on a non-element object.

Reply:
Have made a new issue to track this and added an inline issue in the spec referencing it. https://github.com/w3c/trusted-types/issues/520